File: src/main/java/com/qtech/bubbles/common/entity/Attribute.java
package com.qtech.bubbles.common.entity;

import java.util.HashMap;

public class Attribute {
    private static final HashMap<String, Attribute> attributeMap = new HashMap<>();

    public static final Attribute SCORE_MULTIPLIER = new Attribute("bubbleblaster.score_multiplier");
    public static final Attribute MAX_DAMAGE = new Attribute("bubbleblaster.max_health");
    public static final Attribute DEFENSE = new Attribute("bubbleblaster.defense");
    public static final Attribute ATTACK = new Attribute("bubbleblaster.attack");
    public static final Attribute SPEED = new Attribute("bubbleblaster.speed");
    public static final Attribute LUCK = new Attribute("bubbleblaster.luck");

    private final String name;

    public Attribute(String name) {
        if (Attribute.attributeMap.containsKey(name)) {
            throw new IllegalArgumentException("Duplicate attribute detected!");
        }

        Attribute.attributeMap.put(name, this);
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static Attribute fromName(String name) {
        return Attribute.attributeMap.get(name);
    }
}

File: src/main/kotlin/com/qtech/bubbles/common/entity/Attribute.kt
package com.qtech.bubbles.common.entity

class Attribute(name: String) {
    val name: String

    companion object {
        private val attributeMap = HashMap<String, Attribute>()
        @JvmField
        val SCORE_MULTIPLIER = Attribute("bubbleblaster.score_multiplier")
        @JvmField
        val MAX_DAMAGE = Attribute("bubbleblaster.max_health")
        @JvmField
        val DEFENSE = Attribute("bubbleblaster.defense")
        @JvmField
        val ATTACK = Attribute("bubbleblaster.attack")
        @JvmField
        val SPEED = Attribute("bubbleblaster.speed")
        @JvmField
        val LUCK = Attribute("bubbleblaster.luck")
        @JvmStatic
        fun fromName(name: String): Attribute? {
            return attributeMap[name]
        }
    }

    init {
        require(!attributeMap.containsKey(name)) { "Duplicate attribute detected!" }
        attributeMap[name] = this
        this.name = name
    }
}